core/unit_test/hip/TestHIP_InterOp_StreamsMultiGPU.cpp

```diff
@@ -0,0 +1,108 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <TestHIP_Category.hpp>
+#include <TestMultiGPU.hpp>
+
+namespace {
+
+struct StreamsAndDevices {
+  std::array<hipStream_t, 2> streams;
+  std::array<int, 2> devices;
+
+  StreamsAndDevices() {
+    int n_devices;
+    KOKKOS_IMPL_HIP_SAFE_CALL(hipGetDeviceCount(&n_devices));
+
+    devices = {0, n_devices - 1};
+    for (int i = 0; i < 2; ++i) {
+      KOKKOS_IMPL_HIP_SAFE_CALL(hipSetDevice(devices[i]));
+      KOKKOS_IMPL_HIP_SAFE_CALL(hipStreamCreate(&streams[i]));
+    }
+  }
+  StreamsAndDevices(const StreamsAndDevices &)            = delete;
+  StreamsAndDevices &operator=(const StreamsAndDevices &) = delete;
+  ~StreamsAndDevices() {
+    for (int i = 0; i < 2; ++i) {
+      KOKKOS_IMPL_HIP_SAFE_CALL(hipSetDevice(devices[i]));
+      KOKKOS_IMPL_HIP_SAFE_CALL(hipStreamDestroy(streams[i]));
+    }
+  }
+};
+
+std::array<TEST_EXECSPACE, 2> get_execution_spaces(
+    const StreamsAndDevices &streams_and_devices) {
+  TEST_EXECSPACE exec0(streams_and_devices.streams[0]);
+  TEST_EXECSPACE exec1(streams_and_devices.streams[1]);
+
+  // Must return void to use ASSERT_EQ
+  [&]() {
```

Comment:
Why do you create a lambda function here?

Reply:
`ASSERT_EQ` must be enclosed in a function that returns void. I changed to using `EXPECT_EQ`.